base/linalg/dense.jl

```diff
@@ -29,6 +29,21 @@ function scale!(X::Array{T}, s::Real) where T<:BlasComplex
     X
 end
 
+function isone(x::StridedMatrix)
+    m, n = size(x)
+    m != n && return false # only square matrices can satisfy x == one(x)
+    for i in 1:m
```

Comment:
Perhaps put `@inbounds` before the loop, and also put the loops on the same line, i.e.

```julia
@inbounds for i in 1:m, j in i:m
    [...]
end
```